### PR TITLE
Enhance audit logging: Support comma-separated header filtering with comprehensive tests

### PR DESCRIPTION
**Pull request by suvodeep-pyne**
## Summary
- Transform captureRequestHeaders configuration from boolean to comma-separated list of specific headers to capture
- Implement case-insensitive header matching per HTTP RFC 7230
- Add comprehensive test coverage for edge cases and various string formats
- Refactor header filtering logic to reuse existing toMap function with allow list parameter

## Changes
- **AuditConfig**: Change `_captureRequestHeaders` from `boolean` to `String`
- **AuditRequestProcessor**: Add `parseAllowedHeaders()` utility and enhance `toMap()` with filtering
- **Tests**: Add comprehensive test coverage for configuration parsing, case sensitivity, value handling, and end-to-end flow

## Test Coverage
- Configuration parsing edge cases (empty, whitespace, malformed comma separation)
- Case-insensitive HTTP header matching
- Single vs multiple header values handling
- Complete request processing integration tests

## Breaking Changes
None - this is an unlaunched feature in active development.

### CODE REVIEW

File: pinot-common/src/main/java/org/apache/pinot/common/audit/AuditConfig.java [hunk 1]
```diff
@@ -36,7 +36,7 @@ public final class AuditConfig {
   private boolean _captureRequestPayload = false;
 
   @JsonProperty("capture.request.headers")
-  private boolean _captureRequestHeaders = false;
+  private String _captureRequestHeaders = "";
```
Comment (somandal) [inline pinot-common/src/main/java/org/apache/pinot/common/audit/AuditConfig.java:39]: nit: should this be renamed to `allowedRequestHeaders` instead since now it isn't a boolean but a string? (same for the JsonProperty and the setter / getter / log)

File: pinot-common/src/main/java/org/apache/pinot/common/audit/AuditRequestProcessor.java [hunk 2]
```diff
@@ -140,10 +165,15 @@ private AuditEvent.AuditRequestPayload captureRequestPayload(ContainerRequestCon
...
-          payload.setHeaders(toMap(headers));
+
+      Set<String> allowedHeaders = parseAllowedHeaders(config.getCaptureRequestHeaders());
+      if (!allowedHeaders.isEmpty()) {
+        MultivaluedMap<String, String> allHeaders = requestContext.getHeaders();
+        if (!allHeaders.isEmpty()) {
+          Map<String, Object> filteredHeaders = toMap(allHeaders, allowedHeaders);
+          if (!filteredHeaders.isEmpty()) {
+            payload.setHeaders(filteredHeaders);
+          }
```
Comment (somandal) [inline pinot-common/src/main/java/org/apache/pinot/common/audit/AuditRequestProcessor.java:176]: is there a use case / scenario where we may want to capture all headers? should wildcard `*` be allowed?

**Review by somandal — APPROVED** [review]
mostly lgtm, some minor comments / questions

File: pinot-common/src/main/java/org/apache/pinot/common/audit/AuditRequestProcessor.java (hunk 2 above)
Comment (suvodeep-pyne) [reply pinot-common/src/main/java/org/apache/pinot/common/audit/AuditRequestProcessor.java:176]: I actually had that earlier but seems like it is not supported in OTEL. When I was testing it out, it captures a ton of data including auth headers which is a security risk. Hence, removed that capability and wildcard support.

File: pinot-common/src/main/java/org/apache/pinot/common/audit/AuditConfig.java (hunk 1 above)
Comment (suvodeep-pyne) [reply pinot-common/src/main/java/org/apache/pinot/common/audit/AuditConfig.java:39]: agree in a way. I just thought of keeping it consistent with OTEL standard.
https://opentelemetry.io/docs/zero-code/java/agent/instrumentation/http/

Yes. This is a breaking change where it is now a list of allowed headers. 

For booleans, I am using a suffix like `.enabled` to denote that.
example: `capture.request.payload.enabled`

I have also posted this for further feedback.

Comment (suvodeep-pyne) [reply pinot-common/src/main/java/org/apache/pinot/common/audit/AuditConfig.java:39]: Also, the getters/setters are already updated btw.

Comment (somandal) [reply pinot-common/src/main/java/org/apache/pinot/common/audit/AuditConfig.java:39]: got it, we can keep it as is then
